### PR TITLE
fix: handle obsolete where filter fields mapping to current schema

### PR DESCRIPTION
**Pull request by Junjiequan**
## Description
This PR adds a new function, convertObsoleteWhereFilterToCurrentSchema, to handle mapping obsolete filter fields (e.g., proposalId → proposalIds, sampleId → sampleIds) to their current schema equivalents. 

## Fixes

* Bug fixed (#X)

## Changes:

* changes made

## Tests included

- [ ] Included for each change/fix?
- [ ] Passing? 

## Documentation
- [ ] swagger documentation updated (required for API changes)
- [ ] official documentation updated

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -64,7 +64,6 @@ OIDC_USERQUERY_OPERATOR=<"or"|"and">
 OIDC_USERQUERY_FILTER="username:username, email:email"
 
 ELASTICSEARCH_ENABLED=<"yes"|"no">
-APP_PORT=3003
```
Comment (bpedersen2) [inline .env.example:67]: nit: unrelated?

**Review by bpedersen2 — APPROVED** [review]

File: .env.example (hunk 1 above)
Comment (Junjiequan) [reply .env.example:67]: This ENV variable is not used anywhere and I felt the change was too small to open a PR..  As it's an unused variable in an example file

**Review by nitrosx — APPROVED** [review]
